src/wallet/connection.ts

```diff
@@ -88,4 +91,85 @@ class WalletServiceConnection extends BaseConnection {
   }
 }
 
-export default WalletServiceConnection;
+/**
+ * This connection is used when the wallet has websocket disabled.
+ */
+export class DummyWalletServiceConnection implements IConnection {
+  public readonly isDummyConnection = true;
+  public walletId?: string;
+
+  private currentNetwork: string;
+  private currentServer: string;
+  protected state: ConnectionState;
+
+  constructor(options: WalletServiceConnectionParams = { walletId: 'dummy-wallet' }) {
+    const {
+      walletId,
+      network,
+      servers,
+    } = {
+      ...DEFAULT_PARAMS,
+      ...options,
+    };
+
+    if (!network) {
+      throw Error('You must explicitly provide the network.');
+    }
+
+    this.walletId = walletId;
+    this.currentNetwork = network || 'dummynet';
+    this.state = ConnectionState.CLOSED;
+    this.currentServer = servers[0] || config.getServerUrl();
+  }
+
+  /**
+   * Dummy method to simulate the EventEmitter.on method to avoid errors
+   * @param _event dummy event
+   * @param _listener dummy listener
+   * @returns dummy emitter
+   */
+  on(_event: string, _listener: (...args: any[]) => void): EventEmitter {
```

Comment:
If you need to implement this method, maybe we should continue extending `EventEmitter`, right? And then we can just remove this method from here. 

Is that correct?

Reply:
Correct. But with this dummy EventEmitter we can avoid side effects by guaranteeing it will do nothing, even if the wallet installs any listener on the connection.